packages/react-native-platform-override/src/Override.ts

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import * as Serialized from './Serialized';
+import * as path from 'path';
+
+import UpgradeStrategy, {UpgradeStrategies} from './UpgradeStrategy';
+import ValidationStrategy, {ValidationStrategies} from './ValidationStrategy';
+import OverrideFactory from './OverrideFactory';
+
+/**
+ * Immmutable programatic representation of an override. This should remain
+ * generic to files vs directories, different representations, different
+ * validation rules, etc.
+ */
+export default interface Override {
+  /**
+   * Identifer of the override (e.g. filename or directory name)
+   */
+  name(): string;
```

Comment:
Curious why you are using functions for readonly properties. TypeScript does have support for readonly.
```suggestion
  readonly name: string;
```
I can totally understand if this has to operate and give readonly guarantee when operating with regular JS.

Reply:
We don't care much about untyped scenarios, but this is mostly for consistency with other methods. Not sure if how inheritance stuff would work for ValidationStrategies if we used props.